product/views.py

```diff
@@ -1,44 +1,83 @@
-            return JsonResponse({'error' : "DRINK_DOES_NOT_EXIST"}, status=400)
+import random
+import numpy as np
+
+from django.views import View
+from django.http  import JsonResponse, HttpResponse
+
+from sklearn.preprocessing   import OrdinalEncoder
+from sklearn.naive_bayes     import CategoricalNB
+
+from .utils      import classifier, input_file
+from user.utils  import check_user_slider
+from user.models import User
+from .models     import (
+    MainCategory, 
+    SubCategory,
+    Drink,
+    Allergy,
+    DrinkAllergy,
+    Size,
+    Nutrition,
+    DrinkStatus
+)
+
+
+class DrinkListView(View):
+    def get(self, request):
+        names            = request.GET.get("name", None)
+        filter_set       = {}
+
+        if names:
+            filter_set['name__in'] = names.split(',')
+           
+        subcategory_list = [
+        {
+        'subcategory_id' : subcategory.id,
+        'name'           : subcategory.name,
+        'description'    : subcategory.description,
+        'products' : [
+            {
+                'id'         : drink.id,
+                'isnew'      : drink.is_new,
+                'isseason'   : drink.is_season,
+                'title'      : drink.korean_name,
+                'image'      : drink.image_url,
+                'size'       : drink.nutritions.first().size.name,
+                'kcal'       : drink.nutritions.first().kcal,
+                'sodium'     : drink.nutritions.first().sodium,
+                'saturation' : drink.nutritions.first().saturation,
+                'sugar'      : drink.nutritions.first().sugar,
+                'protein'    : drink.nutritions.first().protein,
+                'caffeine'   : drink.nutritions.first().caffeine
+            } for drink in subcategory.drinks.all()] 
+        }for subcategory in SubCategory.objects.prefetch_related('drinks__nutritions').filter(**filter_set)]
+
+        return JsonResponse({"results" : subcategory_list}, status = 200)
+
+class MainSlideProductsView(View):
+
+    @check_user_slider
+    def get(self, request):
+        if request.user:
+            user = request.user
+            clf  = classifier(input_file)
+
+            enc = OrdinalEncoder()
+            X   = np.array([[user.age, user.gender]])
+            enc.fit(X)
+            OrdinalEncoder()
+            X   = enc.transform(X)
+
+            subcategory_predicted = clf.predict(X)[0]
+            
+            drinks = Drink.objects.filter(sub_category=SubCategory.objects.get(id=subcategory_predicted+1))
+            urls   = [drink.image_url for drink in drinks]
+            return JsonResponse({
+                "my_urls": urls
+            }, status=200)
+        else:
+            drinks = Drink.objects.filter(sub_category=SubCategory.objects.get(id=random.randint(1, 9)))
+            urls   = [drink.image_url for drink in drinks]
+            return JsonResponse({
+                "my_urls": urls 
+            }, status=200)
```

Comment:
```suggestion
        
        drinks = Drink.objects.filter(sub_category=SubCategory.objects.get(id=random.randint(1, 9)))
        urls   = [drink.image_url for drink in drinks]
        return JsonResponse({"my_urls": urls}, status=200)
```

위 if 문의 `return`이 있기 때문에 불필요한 `else` 구문입니다 :)

Reply:
:)